dll/win32/shell32/wine/control.c

```diff
@@ -821,26 +821,29 @@ Control_ShowAppletInTaskbar(CPlApplet* applet, UINT index)
 {
     ITaskbarList* pTaskbar = NULL;
 
-    SetWindowTextW(applet->hWnd, applet->info[index].name);
-    SendMessageW(applet->hWnd, WM_SETICON, ICON_SMALL, (LPARAM)applet->info[index].icon);
-
-    /* Add button to the taskbar */
-    ShowWindow(applet->hWnd, SW_SHOWMINNOACTIVE);
-
-    /* Activate the corresponding button in the taskbar */
-    CoInitialize(NULL);
-    if (CoCreateInstance(&CLSID_TaskbarList,
-                         NULL, CLSCTX_INPROC_SERVER,
-                         &IID_ITaskbarList,
-                         (LPVOID*)&pTaskbar) == S_OK)
+    if (GetParent(applet->hWnd) == NULL) // Is desktop the parent window?
     {
-        if (ITaskbarList_HrInit(pTaskbar) == S_OK)
+        SetWindowTextW(applet->hWnd, applet->info[index].name);
+        SendMessageW(applet->hWnd, WM_SETICON, ICON_SMALL, (LPARAM)applet->info[index].icon);
+
```

Comment:
You may be interested in investigating
https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-itaskbarlist-addtab

Reply:
About the idea of using the method ITaskbarList::AddTab:

What happens if the explorer crashes and restarts automatically?
Then with this method the switch in the taskbar is gone.

With my method, the switches of all open applications including the system control applets are restored.

I can implement the idea to use the return value of ITaskbarList_HrInit as condition to add.

Reply:
> What happens if the explorer crashes and restarts automatically?
Then with this method the switch in the taskbar is gone.

Ah yes, good point.

Reply:
> > What happens if the explorer crashes and restarts automatically?
> > Then with this method the switch in the taskbar is gone.
> 
> Ah yes, good point. But the same sort of observation could be said about the following: with this PR, start a control panel while explorer is not running, then start the explorer. Would the applet show in the taskbar? How does this behave on Windows?

Yes, and Win 7 also reconstructs the switch.

I have uploaded a video. [CORE-18175](https://jira.reactos.org/browse/CORE-18175)

Reply:
The idea to use the return value of ITaskbarList_HrInit is not so good after all.

The following scenario:
- Taskbar exists
- an application uses a system control applet like the installer. (application window = RunDLL parent window)

Result: visual glitch in application, but no button in taskbar